### PR TITLE
Refactor router

### PR DESCRIPTION
**Pull request by simonnguyen3054**
- refactor routing
- add deal item page
- style matched deals page

Note: More refactoring for deals component to pass props to deal item component

### CODE REVIEW

File: client/src/App.js [hunk 1]
```diff
@@ -15,19 +16,22 @@ import "./App.css";
 class App extends Component {
   render() {
     return (
-      <Router>
+      <Switch>
```
Comment (avakrishn) [inline client/src/App.js]: @simonnguyen3054 Just wondering why you changed the tag from `<Router>` to `<Switch>`?

File: client/src/components/Feed/Deals/DealItem/DealItem.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    super();
+
+    this.state = {
+      dealItem: []
+    };
+  }
+
+  async componentDidMount () {
+    //return the param value
+    const { deal_name } = this.props.match.params;
```
Comment (avakrishn) [inline client/src/components/Feed/Deals/DealItem/DealItem.js]: @simonnguyen3054 Can you explain the line `const { deal_name } = this.props.match.params` ?

File: client/src/components/Feed/Deals/Deals.js [hunk 3]
```diff
@@ -82,18 +87,36 @@ class Venues extends Component {
...
-                  <div className="card">
-                    <div className="card-body">
-                      <img
-                        className="card-img-top"
-                        src={deal.deal_image}
-                        alt="deal"
-                      />
-                      <h5 className="card-title">{deal.deal_name}</h5>
+              <div key={deal.id} className="deal">
+                <Link to={`/feed/deals/${deal.deal_name}`} style={{ textDecoration: 'none', color: "black" }} >
```
Comment (avakrishn) [inline client/src/components/Feed/Deals/Deals.js]: @simonnguyen3054 I really like how you are doing the routing! 
Does everything inside the `<Link>` tags become the "link"? In other words can you click on any element inside the Link tags on the page and be taken to the new route?

File: client/src/components/Profile/UserProfile/UserProfile.js [hunk 4]
```diff
@@ -226,7 +226,7 @@ class UserProfile extends Component {
   render() {
     console.log(this.state);
     // console.log(this.props.location.pathname);
-    // console.log(this.props.match.params); 
+    // console.log(this.props.match.params);
```
Comment (avakrishn) [inline client/src/components/Profile/UserProfile/UserProfile.js]: @simonnguyen3054 What was the change that you made here?

File: client/src/App.js (hunk 1 above)
Comment (simonnguyen3054) [reply client/src/App.js]: so Switch just to group the relevant routes together. It's not really make a difference in this case but I did it anyway so later we can use it.

File: client/src/components/Feed/Deals/DealItem/DealItem.js (hunk 2 above)
Comment (simonnguyen3054) [reply client/src/components/Feed/Deals/DealItem/DealItem.js]: This is just object destructuring @avakrishn 

Anytime now when I use deal_name variable, I'm actually passing the value. So in this case, 
```const dealItemArr = await fetch(`http://localhost:3001/api/deals/${deal_name}`); ```

${deal_name} is the value

File: client/src/components/Profile/UserProfile/UserProfile.js (hunk 4 above)
Comment (simonnguyen3054) [reply client/src/components/Profile/UserProfile/UserProfile.js]: ohh there shouldn't be any change, I was just looking at your state refactoring to services. Maybe I added a space accidently

**Pull request merged**